sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -322,11 +322,6 @@ object SQLConf {
       .intConf
       .createWithDefault(4000)
 
-  val PARTITION_DISCOVERY_ENABLED = SQLConfigBuilder("spark.sql.sources.partitionDiscovery.enabled")
```

Comment:
It sounds like we still keep another SQLConf `PARALLEL_PARTITION_DISCOVERY_THRESHOLD`. However, the description of `PARALLEL_PARTITION_DISCOVERY_THRESHOLD` is not right after the refactoring. Could you update the descriptions too?